stdlib/universe/difference.go

```diff
@@ -184,12 +197,12 @@ func (t *differenceTransformation) Process(id execute.DatasetID, tbl flux.Table)
 		}); err != nil {
 			return err
 		}
-		differences[j] = newDifference(t.nonNegative)
+		differences[j] = newDifference(t.nonNegative, t.initialZero)
 	}
 
 	// We need to drop the first row since its difference is undefined
 	firstIdx := 1
-	if t.keepFirst {
+	if t.keepFirst || t.initialZero {
```

Comment:
This seems incorrect. The initial zero does not imply keep first. Initial zero changes the behavior of keep first if it is specified, but otherwise does nothing if keep first is false.

Reply:
I have implemented it in such a way that both are independent of each other. If both are true then initialZero will take priority. Kindly check this test case - https://github.com/influxdata/flux/blob/24dbe5474595fb8420990b80c49f8debfa4772e0/stdlib/universe/difference_test.go#L970

Kindly let me know if you want me to make initialZero dependent on keepFirst.

Reply:
I would prefer if they were independent of each other. The implementation seems to do the opposite. My understanding is that if initial zero is true and keep first is false, the first row will _not_ be materialized. This seems to always materialize the first row as zero whenever initial zero is specified.

As an example, imagine an input of `[20, 30, 50, 30, 40]`. I'd expect the following outputs given the parameters:

```
(keepFirst: false, initialZero: false) -> [10, 20, null, 10]
(keepFirst: true, initialZero: false) -> [null, 10, 20, null, 10]
(keepFirst: false, initialZero: true) -> [10, 20, 30, 10]
(keepFirst: true, initialZero: true) -> [0, 10, 20, 30, 10]
```

Reply:
I have updated the code accordingly and added the test cases for all 4 combinations.